public/app/features/query/state/updateQueries.ts

```diff
@@ -1,27 +1,55 @@
-import { DataQuery, DataSourceInstanceSettings, getDataSourceRef } from '@grafana/data';
+import {
+  DataQuery,
+  DataSourceInstanceSettings,
+  getDataSourceRef,
+  hasQueryExportSupport,
+  hasQueryImportSupport,
+} from '@grafana/data';
+import { getDataSourceSrv } from '@grafana/runtime';
 import { isExpressionReference } from '@grafana/runtime/src/utils/DataSourceWithBackend';
 
-export function updateQueries(
+export async function updateQueries(
   newSettings: DataSourceInstanceSettings,
   queries: DataQuery[],
   dsSettings?: DataSourceInstanceSettings
-): DataQuery[] {
+): Promise<DataQuery[]> {
+  let nextQueries = queries;
   const datasource = getDataSourceRef(newSettings);
+  const nextDS = await getDataSourceSrv().get(newSettings.uid);
```

Comment:
looks like we make this call twice now.  maybe we can pass the ds in to avoid this?

Reply:
fixed now :)